Add release notes for 22.05.0. (#1607)
o3de/o3de.org: content/docs/release-notes/archive/22-05-0/_index.md

  * Several improvements to Diffuse GI and ReflectionProbe visual quality.
* **White box**
  * White box is now supported on Linux! ([5075](https://github.com/o3de/o3de/pull/5075))
 
## Deprecations
 
* **Physics**
    * `TransformUniformScale` ([7573](https://github.com/o3de/o3de/issues/7573))
    * `BoxManipulatorRequestBus` ([7572](https://github.com/o3de/o3de/issues/7572))
* **EMFX (Character)**
    * OpenGL Render Plugin is now deprecated and replaced by EMFX Atom Viewport.
 
## Known issues
 
* **EMFX (Character)**
    * If you have saved a layout in the old viewport, it will load the old OpenGL viewport. A workaround is to delete the OpenGL widget and add a new Atom viewport widget, then save the layout.  Alternatively, you can recreate the same layout from the default layout.
 
* **Atom (Graphics)**
    * [Linux] Editor crashes when pressing the **Activate LUT** button right after generating LUT in the HDR Color Grading component. ([9157](https://github.com/o3de/o3de/issues/9157))
 
* **Asset Processor**
    * [Linux]: Closing Editor while Asset Processor is still running will result in a crash upon relaunching Editor.

> * In rare situations, on Linux Ubuntu 20.04.4 LTS, it might not be possible to launch O3DE Editor, Project Manager, Asset Processor, or other executables from the file explorer, and an error is displayed.

Reviewer: ```suggestion
 * **Linux Ubuntu 20.04.4 LTS**
     In rare situations, on Linux Ubuntu 20.04.4 LTS, it might not be possible to launch O3DE Editor, Project Manager, Asset Processor, or other executables from the file explorer, and an error is displayed.
```

  * Several improvements to Diffuse GI and ReflectionProbe visual quality.
* **White box**
  * White box is now supported on Linux! ([5075](https://github.com/o3de/o3de/pull/5075))
 
## Deprecations
 
* **Physics**
    * `TransformUniformScale` ([7573](https://github.com/o3de/o3de/issues/7573))
    * `BoxManipulatorRequestBus` ([7572](https://github.com/o3de/o3de/issues/7572))
* **EMFX (Character)**
    * OpenGL Render Plugin is now deprecated and replaced by EMFX Atom Viewport.
 
## Known issues
 
* **EMFX (Character)**
    * If you have saved a layout in the old viewport, it will load the old OpenGL viewport. A workaround is to delete the OpenGL widget and add a new Atom viewport widget, then save the layout.  Alternatively, you can recreate the same layout from the default layout.
 
* **Atom (Graphics)**
    * [Linux] Editor crashes when pressing the **Activate LUT** button right after generating LUT in the HDR Color Grading component. ([9157](https://github.com/o3de/o3de/issues/9157))
 
* **Asset Processor**
    * [Linux]: Closing Editor while Asset Processor is still running will result in a crash upon relaunching Editor.

* **Linux Ubuntu 20.04.4 LTS**

>      In rare situations, on Linux Ubuntu 20.04.4 LTS, it might not be possible to launch O3DE Editor, Project Manager, Asset Processor, or other executables from the file explorer, and an error is displayed.

Reviewer: third time's a charm

```suggestion

    In rare situations, on Linux Ubuntu 20.04.4 LTS, it might not be possible to launch O3DE Editor, Project Manager, Asset Processor, or other executables from the file explorer, and an error is displayed.
```

 
* **Physics**
    * `TransformUniformScale` ([7573](https://github.com/o3de/o3de/issues/7573))
    * `BoxManipulatorRequestBus` ([7572](https://github.com/o3de/o3de/issues/7572))
* **EMFX (Character)**
    * OpenGL Render Plugin is now deprecated and replaced by EMFX Atom Viewport.
 
## Known issues
 
* **EMFX (Character)**
    * If you have saved a layout in the old viewport, it will load the old OpenGL viewport. A workaround is to delete the OpenGL widget and add a new Atom viewport widget, then save the layout.  Alternatively, you can recreate the same layout from the default layout.
 
* **Atom (Graphics)**
    * [Linux] Editor crashes when pressing the **Activate LUT** button right after generating LUT in the HDR Color Grading component. ([9157](https://github.com/o3de/o3de/issues/9157))
 
* **Asset Processor**
    * [Linux]: Closing Editor while Asset Processor is still running will result in a crash upon relaunching Editor.

* **Linux Ubuntu 20.04.4 LTS**

    In rare situations, on Linux Ubuntu 20.04.4 LTS, it might not be possible to launch O3DE Editor, Project Manager, Asset Processor, or other executables from the file explorer, and an error is displayed.

>     **Workaround:** Use the command line to launch the executable from its directory, for example `sudo ./o3de`. Refer to [issue 9502](https://github.com/o3de/o3de/issues/9502) for more infrormation.

Reviewer: Additional info request:

```suggestion
    Workaround: Use the command line to launch the executable from its directory, for example `sudo ./o3de`. Refer to this [FOSTips article](https://fostips.com/double-click-run-elf-ubuntu/) for other possible solutions.
    
    Refer to [issue 9502](https://github.com/o3de/o3de/issues/9502) for more information.
```

  * Upgrade Physics Automated Tests:All physics automated tests have been updated to use Prefabs instead of Slices (including PhysX, Cloth and Blast). All tests are using the new test framework (in most cases). Several are also now running in parallel.
* **Installer**
  * Installer validation tests: These tests verify that an installer build is valid. The automated tests verify that key binaries exist, that o3de.exe registers engine, create project succeeds, compile project succeeds, asset processor batch succeeds, editor runs, game launcher runs, uninstall succeeds and unregisters engine. Currently run as nightly for O3DE, though the tests are designed so anyone can plug these tests into their quality verification process. ([7834](https://github.com/o3de/o3de/pull/7834))
* **Script Canvas**
  * Script Canvas now has a "mini map" feature that helps visualize and navigate graphs. ([6263](https://github.com/o3de/o3de/pull/6263))

> * **Viewport**

Reviewer 2: Adding late, approved addition to the release notes (default component template).
Sig-release PR: https://github.com/o3de/sig-release/pull/56
```suggestion
* **Templates**
  * Default component template: A new template is available to help you create components. Use the `o3de` CLI script and the `create-from-template` command to start a runtime component from the **DefaultComponent** template. For detailed instructions refer to [Creating a Component in O3DE](/docs/user-guide/programming/components/create-component/). ([8462](https://github.com/o3de/o3de/pull/8462))
* **Viewport**
```

  * Prefabs Focus Mode Improvements: A number of fixes and quality of life improvements have been added to Focus Mode Prefab editing. Now **Ctrl-S** will save the focused Prefab not the entire scene, Focus Mode breadcrumbs now have icons, the right-click context menu grouping has been improved, as well as a number of other small fixes and updates.
  * Cursor wrap mode for manipulators: The cursor can now be set to wrap around the viewport during navigation in the Editor. This feature is off by default and can be turned on in the settings registry with `--regset=/Amazon/Preferences/Editor/Manipulator/ManipulatorMouseWrapSetting=true`. ([5155](https://github.com/o3de/o3de/pull/5155))
* **Atom**
  * New Debug Rendering Level Component: There's a new level component that allows users to visualize and debug various aspects of the rendering, for example visualize the normals, show only diffuse or specular lighting, override certain properties for all materials in the scene, and more. 
  * Pipeline Global Pass Attachment Reference: Users writing `.pass` files for rendering can now declare and reference attachments in a way that is global to the pipeline.

>   * Usability and performance improvements for `DiffuseProbeGrid`:

Reviewer 3: ```suggestion
  * Usability and performance improvements for DiffuseProbeGrid:
```

---

> linkTitle: 22.05.0 Release notes
> sectionPageTitle: Release notes
> title: Release notes for Open 3D Engine 22.05.0

Reviewer 2: ```suggestion
linkTitle: 22.05.0 Release Notes
sectionPageTitle: Release Notes
title: Release Notes for Open 3D Engine 22.05.0
```

description: Full release notes for Open 3D Engine (O3DE) version 22.05.0.
weight: 897
toc: true
---

**Open 3D Engine (O3DE)** version 22.05.0 represents the project's first major release of 2022. This release has seen 1,417 code submissions that contain bug fixes, quality of life improvements, and feature additions. Starting with this release, our release notes will include a [feature grid](./feature-state/) representing the current state of development for each feature in O3DE.

## Highlights for 22.05.0

First of all: Open 3D Engine has a new logo! Our [branding guidelines](https://www.o3de.org/docs/tools-ui/branding-guidelines/) have been updated.

> This release also sees the introduction of User Defined Properties (UDP), a way to read metadata from source assets into the Asset processor. UDP can be assigned in content creation tools to store custom properties about hierarchy nodes such as mesh, light, animation nodes, etc., to power asset generation workflows for O3DE. For more information, read our [blog post on UDP](https://docs.o3de.org/blog/posts/blog-udp/).

Reviewer 2: ```suggestion
This release also sees the introduction of User Defined Properties (UDP), a way to read metadata from source assets into the Asset Processor. UDP can be assigned in content creation tools to store custom properties about hierarchy nodes such as mesh, light, animation nodes, etc., to power asset generation workflows for O3DE. For more information, read our [blog post on UDP](https://docs.o3de.org/blog/posts/blog-udp/).
```

> We've also included an experimental Gem in release for motion matching. Motion matching is a data-driven animation technique that synthesizes motions based on existing animation data and the current character and input contexts. The example Gem includes a character prefab, controllable using a gamepad. Find more details on the supported features and how things work internally in the [Motion Matching Gem code](https://github.com/o3de/o3de/tree/development/Gems/MotionMatching) in our GitHub repository.

Reviewer: ```suggestion
We've also included an experimental Gem in this release for motion matching. Motion matching is a data-driven animation technique that synthesizes motions based on existing animation data and the current actor and input contexts. The example Gem includes a character prefab, controllable using a gamepad. Find more details on the supported features and how things work internally in the [Motion Matching Gem code](https://github.com/o3de/o3de/tree/development/Gems/MotionMatching) in our GitHub repository.
```

    * DiffuseProbeGrid Irradiance query: Query the diffuse GI at any position and direction in the level. Improved DiffuseProbeGrid blending when multiple grids overlap and around the edges of a grid to blend with the global IBL cubemap.
  * New **CubeMapCapture** component: Captures a diffuse or specular convolved cubemap at any location in the level.
  * The MSAA state can now be changed in the `MainRenderPipeline.azasset` file by setting the `MultisampleState/samples` parameter.  Setting the samples to `1` will disable MSAA.
  * Several improvements to Diffuse GI and ReflectionProbe visual quality.
* **White box**
  * White box is now supported on Linux! ([5075](https://github.com/o3de/o3de/pull/5075))
 
## Deprecations
 
* **Physics**
    * `TransformUniformScale` ([7573](https://github.com/o3de/o3de/issues/7573))
    * `BoxManipulatorRequestBus` ([7572](https://github.com/o3de/o3de/issues/7572))
* **EMFX (Character)**
    * OpenGL Render Plugin is now deprecated and replaced by EMFX Atom Viewport.
 

> ## Known Issues

Reviewer 2: ```suggestion
## Known issues
```

 
* **EMFX (Character)**
    * If you have saved a layout in the old viewport, it will load the old OpenGL viewport. A workaround is to delete the OpenGL widget and add a new Atom viewport widget, then save the layout.  Alternatively, you can recreate the same layout from the default layout.
 
* **Atom (Graphics)**
    * [Linux] Editor crashes when pressing the **Activate LUT** button right after generating LUT in the HDR Color Grading component. ([9157](https://github.com/o3de/o3de/issues/9157))
 
* **Asset Processor**
    * [Linux]: Closing Editor while Asset Processor is still running will result in a crash upon relaunching Editor.

>  

Reviewer: Adding known issue for Linux per team request.

```suggestion

* In rare situations, on Linux Ubuntu 20.04.4 LTS, it might not be possible to launch O3DE Editor, Project Manager, Asset Processor, or other executables from the file explorer, and an error is displayed.

    **Workaround:** Use the command line to launch the executable from its directory, for example `sudo ./o3de`. Refer to [issue 9502](https://github.com/o3de/o3de/issues/9502) for more infrormation.
```

>  

Reviewer: Adding known issue for Linux per team request.

```suggestion

* In rare situations, on Linux Ubuntu 20.04.4 LTS, it might not be possible to launch O3DE Editor, Project Manager, Asset Processor, or other executables from the file explorer, and an error is displayed.

    **Workaround:** Use the command line to launch the executable from its directory, for example `sudo ./o3de`. Refer to [issue 9502](https://github.com/o3de/o3de/issues/9502) for more infrormation.
```
